lib/core/src/msParam.cpp

```diff
@@ -1190,7 +1190,10 @@ namespace
         const auto& bbuf = _func(*execCmdOut);
 
         *_out = static_cast<char*>(std::malloc(bbuf.len + 1));
-        std::memcpy(*_out, static_cast<char*>(bbuf.buf), bbuf.len);
+
+        if (nullptr != bbuf.buf) {
+            std::memcpy(*_out, static_cast<char*>(bbuf.buf), bbuf.len);
+        }
         (*_out)[bbuf.len] = '\0';
```

Comment:
Would it make sense to zero out the structure, or put the `\0` in `(*_out)[0]` if `bbuf.buf == nullptr` (i.e. the `else` case)? As it stands, there could be some garbage string in there since we just `malloc` above.

Reply:
I was assuming if `bbuf.buf` is `nullptr`, then `bbuf.len` is set to zero, but it's better to have the buffer zeroed out. I'll do that.

Reply:
Done.